src/main/java/blue/endless/glow/model/gltf/GLTFLoader.java

```diff
@@ -19,52 +20,65 @@
 import blue.endless.glow.model.gltf.impl.GLTFData;
 
 public class GLTFLoader {
+	private static final Gson GSON = new GsonBuilder()
+		//.registerTypeAdapter(ModelTransformation.class, foo)
+		.create();
+
 	public static Model loadString(String json) throws IOException {
-		GLTFData gltfData = new GsonBuilder()
-				//.registerTypeAdapter(ModelTransformation.class, foo)
-				.create().fromJson(json, GLTFData.class);
-		//GLTFData gltfData = Jankson.builder().build().fromJson(json, GLTFData.class);
-		
+		return loadString(json, null);
+	}
+
+	public static Model loadString(String json, byte[] binaryData) throws IOException {
+		return loadModel(loadRaw(json), binaryData);
+	}
+
+	public static Model loadModel(GLTFData gltfData, byte[] binaryData) throws IOException {
+
 		Model result = new Model();
-		
+
 		Vector3d globalTranslate = new Vector3d(0, 0, 0);
 		if (gltfData.asset.generator != null && gltfData.asset.generator.startsWith("Blockbench")) {
 			globalTranslate = new Vector3d(0, -0.5, 0);
 		}
-		
+
 		for(GLTFData.GLTFMesh mesh : gltfData.meshes) {
 			for (GLTFData.GLTFPrimitive primitive : mesh.primitives) {
-				int[] indexBuffer = gltfData.getScalarAccess(primitive.indices);
-				Vector3d[] positionBuffer = gltfData.getVec3Access(primitive.attributes.POSITION);
-				Vector2d[] uvBuffer = gltfData.getVec2Access(primitive.attributes.TEXCOORD_0);
-				Vector3d[] normalBuffer = gltfData.getVec3Access(primitive.attributes.NORMAL);
+				int[] indexBuffer = gltfData.getScalarAccess(primitive.indices, binaryData);
+				Vector3d[] positionBuffer = gltfData.getVec3Access(primitive.attributes.POSITION, binaryData);
+				Vector2d[] uvBuffer = gltfData.getVec2Access(primitive.attributes.TEXCOORD_0, binaryData);
+				Vector3d[] normalBuffer = gltfData.getVec3Access(primitive.attributes.NORMAL, binaryData);
 				//TODO: look for ColorAccess
-				
```

Comment:
All the whitespace on blank lines is missing, this makes for a HUGE diff. I could ignore some issues but this one is a dealbreaker - code style for the project is to leave whitespace alone when it represents a block indent to the scope it resides in.